app/client/src/components/shared/HmwMap/index.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+
+import React from 'react';
+import type { Node } from 'react';
+import { css } from 'styled-components/macro';
+import Map from '@arcgis/core/Map';
+import MapView from '@arcgis/core/views/MapView';
+import MapWidgets from 'components/shared/MapWidgets';
+import MapMouseEvents from 'components/shared/MapMouseEvents';
+// contexts
+import { LocationSearchContext } from 'contexts/locationSearch';
+
+// --- styled components ---
+const mapContainerStyles = css`
+  position: absolute;
+  width: 100%;
+  height: 100%;
+`;
+
+// --- components ---
+type Props = {
+  layers: Object,
+  startingExtent?: Object,
+  children?: Node,
+};
+
+function HmwMap({ layers = null, startingExtent = null, children }: Props) {
+  const {
+    initialExtent,
+    highlightOptions,
+    getBasemap,
+    mapView,
+    setMapView,
+  } = React.useContext(LocationSearchContext);
+
+  const [map, setMap] = React.useState(null);
+
+  // Initialize the map
+  const [mapInitialized, setMapInitialized] = React.useState(false);
+  React.useEffect(() => {
+    if (mapInitialized) return;
+
+    const newMap = new Map({
+      basemap: getBasemap(),
+      layers: [],
+    });
+    setMap(newMap);
+
+    const view = new MapView({
+      container: 'hmw-map-container',
+      map: newMap,
+      extent: startingExtent ?? initialExtent,
+      highlightOptions,
+    });
+    setMapView(view);
+
+    setMapInitialized(true);
+  }, [
+    mapInitialized,
+    getBasemap,
+    highlightOptions,
+    initialExtent,
+    startingExtent,
+    setMapView,
+  ]);
+
+  return (
+    <>
+      <div id="hmw-map-container" css={mapContainerStyles} />
+
+      {map && mapView && (
+        <>
+          <MapWidgets map={map} view={mapView} layers={layers} />
+
+          <MapMouseEvents map={map} view={mapView} />
+        </>
+      )}
+    </>
```

Comment:
@cschwinderg I'll do a fuller review, but I jumped to this file first... Couldn't you still put the `MapWidgets` and `MapMouseEvents` inside the map container – sort of like we were doing before?

```suggestion
    <div id="hmw-map-container" css={mapContainerStyles}>
      {map && mapView && (
        <>
          <MapWidgets map={map} view={mapView} layers={layers} />
          <MapMouseEvents map={map} view={mapView} />
        </>
      )}
    </div>
```

I just did a quick test, and it seems to work, but wanted to run it by you to confirm. That would remove the wrapping fragment, and simplify the JSX markup slightly.

Reply:
Cool! I didn't think that would work. I went a head and made this change.

Reply:
Great!